### ISSUE TITLE
Crash program when I call getGeners

### ISSUE THREAD

**Issue by o4x** [sc4v3ng3r/flutter_audio_query#22]
I get  this error

```
E/MethodChannel#ryanheise.com/audioServiceBackground( 8672): Failed to handle method call
E/MethodChannel#ryanheise.com/audioServiceBackground( 8672): java.lang.NullPointerException: Attempt to invoke interface method 'void io.flutter.plugin.common.MethodChannel$Result.success(java.lang.Object)' on a null object reference
E/MethodChannel#ryanheise.com/audioServiceBackground( 8672): 	at com.ryanheise.audioservice.AudioServicePlugin.sendStartResult(AudioServicePlugin.java:143)
E/MethodChannel#ryanheise.com/audioServiceBackground( 8672): 	at com.ryanheise.audioservice.AudioServicePlugin.access$800(AudioServicePlugin.java:54)
E/MethodChannel#ryanheise.com/audioServiceBackground( 8672): 	at com.ryanheise.audioservice.AudioServicePlugin$BackgroundHandler.onMethodCall(AudioServicePlugin.java:617)
E/MethodChannel#ryanheise.com/audioServiceBackground( 8672): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:231)
E/MethodChannel#ryanheise.com/audioServiceBackground( 8672): 	at io.flutter.embedding.engine.dart.DartMessenger.handleMessageFromDart(DartMessenger.java:93)
E/MethodChannel#ryanheise.com/audioServiceBackground( 8672): 	at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(FlutterJNI.java:642)
E/MethodChannel#ryanheise.com/audioServiceBackground( 8672): 	at android.os.MessageQueue.nativePollOnce(Native Method)
E/MethodChannel#ryanheise.com/audioServiceBackground( 8672): 	at android.os.MessageQueue.next(MessageQueue.java:376)
E/MethodChannel#ryanheise.com/audioServiceBackground( 8672): 	at android.os.Looper.loop(Looper.java:244)
E/MethodChannel#ryanheise.com/audioServiceBackground( 8672): 	at android.app.ActivityThread.main(ActivityThread.java:6706)
E/MethodChannel#ryanheise.com/audioServiceBackground( 8672): 	at java.lang.reflect.Method.invoke(Native Method)
E/MethodChannel#ryanheise.com/audioServiceBackground( 8672): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
E/MethodChannel#ryanheise.com/audioServiceBackground( 8672): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
E/DartMessenger( 8672): Uncaught exception in binary message listener
E/DartMessenger( 8672): java.lang.IllegalStateException: Reply already submitted
E/DartMessenger( 8672): 	at io.flutter.embedding.engine.dart.DartMessenger$Reply.reply(DartMessenger.java:148)
E/DartMessenger( 8672): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:249)
E/DartMessenger( 8672): 	at io.flutter.embedding.engine.dart.DartMessenger.handleMessageFromDart(DartMessenger.java:93)
E/DartMessenger( 8672): 	at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(FlutterJNI.java:642)
E/DartMessenger( 8672): 	at android.os.MessageQueue.nativePollOnce(Native Method)
E/DartMessenger( 8672): 	at android.os.MessageQueue.next(MessageQueue.java:376)
E/DartMessenger( 8672): 	at android.os.Looper.loop(Looper.java:244)
E/DartMessenger( 8672): 	at android.app.ActivityThread.main(ActivityThread.java:6706)
E/DartMessenger( 8672): 	at java.lang.reflect.Method.invoke(Native Method)
E/DartMessenger( 8672): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
E/DartMessenger( 8672): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
Error -32601 received from application: Method not found
Error -32601 received from application: Method not found
Restarted application in 1,769ms.
method not available: ext.flutter.platformOverride
Reloaded 3 of 757 libraries in 857ms.
Reloaded 3 of 757 libraries in 779ms.

[38;5;248m════════ Exception caught by widgets library ═══════════════════════════════════[39;49m
[38;5;244mThe following assertion was thrown building Library(dirty, state: _LibraryState#71a6f):[39;49m
'dart:ui/painting.dart': Failed assertion: line 188: '<optimized out>': is not true.


[38;5;248mEither the assertion indicates an error in the framework itself, or we should provide substantially more information in this error message to help you determine and fix the underlying cause.
In either case, please report this assertion by filing a bug on GitHub:
  https://github.com/flutter/flutter/issues/new?template=BUG.md
[39;49m

[38;5;244mThe relevant error-causing widget was[39;49m
    [38;5;248mLibrary[39;49m
[38;5;244mWhen the exception was thrown, this was the stack[39;49m
[38;5;244m#2      Color.withOpacity  (dart:ui/painting.dart:188:12)[39;49m
[38;5;248m#3      _LibraryState.build[39;49m
[38;5;244m#4      StatefulElement.build[39;49m
[38;5;244m#5      ComponentElement.performRebuild[39;49m
[38;5;244m#6      Element.rebuild[39;49m
[38;5;244m...[39;49m
[38;5;248m════════════════════════════════════════════════════════════════════════════════[39;49m
Reloaded 3 of 757 libraries in 922ms.
Reloaded 3 of 757 libraries in 877ms.
Reloaded 3 of 757 libraries in 816ms.
E/AndroidRuntime( 8672): FATAL EXCEPTION: AsyncTask #40
E/AndroidRuntime( 8672): Process: com.example.musical, PID: 8672
E/AndroidRuntime( 8672): java.lang.RuntimeException: An error occurred while executing doInBackground()
E/AndroidRuntime( 8672): 	at android.os.AsyncTask$3.done(AsyncTask.java:354)
E/AndroidRuntime( 8672): 	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
E/AndroidRuntime( 8672): 	at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
E/AndroidRuntime( 8672): 	at java.util.concurrent.FutureTask.run(FutureTask.java:271)
E/AndroidRuntime( 8672): 	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245)
E/AndroidRuntime( 8672): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
E/AndroidRuntime( 8672): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
E/AndroidRuntime( 8672): 	at java.lang.Thread.run(Thread.java:764)
E/AndroidRuntime( 8672): Caused by: android.database.sqlite.SQLiteException: no such column: genre_name (code 1 SQLITE_ERROR): , while compiling: SELECT Distinct genre_name FROM audio ORDER BY genre_name ASC
E/AndroidRuntime( 8672): 	at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:179)
E/AndroidRuntime( 8672): 	at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:135)
E/AndroidRuntime( 8672): 	at android.content.ContentProviderProxy.query(ContentProviderNative.java:418)
E/AndroidRuntime( 8672): 	at android.content.ContentResolver.query(ContentResolver.java:804)
E/AndroidRuntime( 8672): 	at android.content.ContentResolver.query(ContentResolver.java:753)
E/AndroidRuntime( 8672): 	at android.content.ContentResolver.query(ContentResolver.java:711)
E/AndroidRuntime( 8672): 	at boaventura.com.devel.br.flutteraudioquery.loaders.GenreLoader$GenreLoadTask.loadData(GenreLoader.java:111)
E/AndroidRuntime( 8672): 	at boaventura.com.devel.br.flutteraudioquery.loaders.GenreLoader$GenreLoadTask.loadData(GenreLoader.java:86)
E/AndroidRuntime( 8672): 	at boaventura.com.devel.br.flutteraudioquery.loaders.tasks.AbstractLoadTask.doInBackground(AbstractLoadTask.java:45)
E/AndroidRuntime( 8672): 	at boaventura.com.devel.br.flutteraudioquery.loaders.tasks.AbstractLoadTask.doInBackground(AbstractLoadTask.java:12)
E/AndroidRuntime( 8672): 	at android.os.AsyncTask$2.call(AsyncTask.java:333)
E/AndroidRuntime( 8672): 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
E/AndroidRuntime( 8672): 	... 4 more
I/Process ( 8672): Sending signal. PID: 8672 SIG: 9
Lost connection to device.
Exited (sigterm)
```

**Comment by olayemii** [comment]
Experienced this too, any fix on this yet?

**Comment by olayemii** [comment]
I have made a little fix for this, apparently, the app crashes because there's no matching song for any genre..
This is my pull request https://github.com/sc4v3ng3r/flutter_audio_query/pull/23 , hope he merges asap

**Comment by o4x** [comment]
Hi @OLayemii , I have tested your fork, the crash is fix but no return any genres.

**Comment by olayemii** [comment]
@o4x yes, that was why i closed my PR, however I have been able to make it work now, I'll push the new codes to my fork now.

**Comment by olayemii** [comment]
You can try to clone and test again, tell me if it works for you

**Comment by o4x** [comment]
thank's very much @OLayemii . I've tested it right now, everything is working fine.
I'm so happy about that .

If you add memberIds property to GenreInfo it would be really great because the method getSongsFromGenre is slow.

**Comment by olayemii** [comment]
@o4x I'm still fixing Alot on the fork, I'll try to optimize all I can as I go along..

**Comment by o4x** [comment]
@OLayemii Bravo .
However, this library has been very helpful to me, but it has many bugs.

do you want merge your fork?

**Comment by olayemii** [comment]
The repo creator isn't accepting PRs, sad enough

**Comment by o4x** [comment]
getSongsFromGenre() return all songs 😂

**Comment by olayemii** [comment]
@o4x 
😂 Bro, I am yet to fix that method, It was crashing like `getGenres()` too..I'll make these updates in a bit.

**Comment by o4x** [comment]
@OLayemii  Bro, Did you find a solution for for getSongsFromGenre() or genreName property?  I really need it and I don't understand anything about Java.

**Comment by olayemii** [comment]
Bro, I'm yet to do it, I'll get back to you this week though

**Comment by o4x** [comment]
@OLayemii , You haven't found a solution yet? Everyone are disabled🙄

**Comment by olayemii** [comment]
Haha, no bro, I havent been able to recheck..any solution yet?

**Comment by o4x** [comment]
I wrote a very simple plugin to just get genres but I haven't merged it with this yet. I put it in my repository although it is useless.

**Comment by olayemii** [comment]
Ok, let me check your implementation @o4x

**Comment by olayemii** [comment]
Seen, that works @o4x ? Why do you say it's useless?

**Comment by o4x** [comment]
Yeah that's work fine. I say it useless because it just get genres.

**Comment by olayemii** [comment]
Okay, the main point of concern is getting songs for genres right?

**Comment by pedrolemoz** [comment]
Same error here, in Android 10. I get the following:

```
E/AndroidRuntime(23721): FATAL EXCEPTION: AsyncTask #1
E/AndroidRuntime(23721): Process: com.pedrolemoz.music_player, PID: 23721
E/AndroidRuntime(23721): java.lang.RuntimeException: An error occurred while executing doInBackground()
E/AndroidRuntime(23721):        at android.os.AsyncTask$4.done(AsyncTask.java:399)
E/AndroidRuntime(23721):        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
E/AndroidRuntime(23721):        at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
E/AndroidRuntime(23721):        at java.util.concurrent.FutureTask.run(FutureTask.java:271)
E/AndroidRuntime(23721):        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:289)
E/AndroidRuntime(23721):        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
E/AndroidRuntime(23721):        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
E/AndroidRuntime(23721):        at java.lang.Thread.run(Thread.java:919)
E/AndroidRuntime(23721): Caused by: android.database.sqlite.SQLiteException: no such column: genre_name (code 1 SQLITE_ERROR): , while compiling: SELECT Distinct genre_name FROM audio WHERE ((is_pending=0) AND (is_trashed=0) AND (volume_name IN ( 'external_primary' ))) ORDER BY genre_name ASC
E/AndroidRuntime(23721):        at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:184)
E/AndroidRuntime(23721):        at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:140)
E/AndroidRuntime(23721):        at android.content.ContentProviderProxy.query(ContentProviderNative.java:423)
E/AndroidRuntime(23721):        at android.content.ContentResolver.query(ContentResolver.java:946)
E/AndroidRuntime(23721):        at android.content.ContentResolver.query(ContentResolver.java:881)
E/AndroidRuntime(23721):        at android.content.ContentResolver.query(ContentResolver.java:837)
E/AndroidRuntime(23721):        at boaventura.com.devel.br.flutteraudioquery.loaders.GenreLoader$GenreLoadTask.loadData(GenreLoader.java:111)
E/AndroidRuntime(23721):        at boaventura.com.devel.br.flutteraudioquery.loaders.GenreLoader$GenreLoadTask.loadData(GenreLoader.java:86)
E/AndroidRuntime(23721):        at boaventura.com.devel.br.flutteraudioquery.loaders.tasks.AbstractLoadTask.doInBackground(AbstractLoadTask.java:45)
E/AndroidRuntime(23721):        at boaventura.com.devel.br.flutteraudioquery.loaders.tasks.AbstractLoadTask.doInBackground(AbstractLoadTask.java:12)
E/AndroidRuntime(23721):        at android.os.AsyncTask$3.call(AsyncTask.java:378)
E/AndroidRuntime(23721):        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
E/AndroidRuntime(23721):        ... 4 more
I/Process (23721): Sending signal. PID: 23721 SIG: 9
Lost connection to device.
```

@sc4v3ng3r Please merge pull requests to this package!

**Comment by sc4v3ng3r** [comment]
solved #37
